sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdConnectionStateListenerMetrics.java

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation.directconnectivity.rntbd;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.time.Instant;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
+@JsonSerialize(using = RntbdConnectionStateListenerMetrics.RntbdConnectionStateListenerMetricsJsonSerializer.class)
+public final class RntbdConnectionStateListenerMetrics implements Serializable {
+    private static final long serialVersionUID = 1L;
+    private static final Logger logger = LoggerFactory.getLogger(RntbdConnectionStateListenerMetrics.class);
+
+    private final AtomicReference<Instant> lastCallTimestamp;
+    private final AtomicReference<Instant> lastActionableTimestamp;
+    private final AtomicInteger lastAddressesUpdatedCount;
+
+    public RntbdConnectionStateListenerMetrics() {
+
+        this.lastCallTimestamp = new AtomicReference<>();
+        this.lastActionableTimestamp = new AtomicReference<>();
+        this.lastAddressesUpdatedCount = new AtomicInteger(0);
+    }
+
+    public void recordAddressUpdated(int addressEntryUpdatedCount) {
+        this.lastActionableTimestamp.set(this.lastCallTimestamp.get());
+        this.lastAddressesUpdatedCount.set(addressEntryUpdatedCount);
+    }
+
+    public void record() {
+        this.lastCallTimestamp.set(Instant.now());
+    }
+
+    final static class RntbdConnectionStateListenerMetricsJsonSerializer extends com.fasterxml.jackson.databind.JsonSerializer<RntbdConnectionStateListenerMetrics> {
+
+        public RntbdConnectionStateListenerMetricsJsonSerializer() {
+        }
+
+        @Override
+        public void serialize(RntbdConnectionStateListenerMetrics metrics, JsonGenerator writer, SerializerProvider serializers) throws IOException {
+            writer.writeStartObject();
+
+            writer.writeStringField(
+                "lastCallTimestamp",
+                metrics.lastCallTimestamp.get() == null ? "N/A" : metrics.lastCallTimestamp.toString());
+
+            if (metrics.lastActionableTimestamp.get() != null) {
+                writer.writeStringField("lastActionableTimestamp", metrics.lastActionableTimestamp.toString());
```

Comment:
since these two are closely related to each other, maybe change to "lastActionableContext": []